File: device-app/android/umsDemo/app/src/main/java/com/rationalowl/umsdemo/domain/ServerRepository.java
package com.rationalowl.umsdemo.domain;

public class ServerRepository extends JsonRepository<Server> {
    private static final String FILE_NAME = "server.json";

    private static ServerRepository instance;

    public static synchronized ServerRepository getInstance() {
        if (instance == null) {
            instance = new ServerRepository();
        }

        return instance;
    }

    private ServerRepository() {
        super(Server.class, FILE_NAME);
    }

    public Server getServer() {
        return getValue();
    }

    public void setServer(Server server) {
        setValue(server);
    }
}

File: device-app/ios-swift/umsDemo_prev/ums/Domain/ServerRepository.swift
class ServerRepository : JsonRepository<Server>
{
    private static let fileName = "server.json"
    
    static let shared = ServerRepository()
    
    private init()
    {
        super.init(ServerRepository.fileName);
    }
    
    var server: Server?
    {
        get
        {
            return getValue()
        }
        set
        {
            setValue(newValue!)
        }
    }
}
